specification/types.md

```diff
@@ -85,3 +86,14 @@ An enumerated error code represented idiomatically in the implementation languag
 A structure containing the following fields:
 
 - hooks (one or more [hooks](./sections/04-hooks.md), optional)
+
+### Provider Events
+
+An enumeration of provider events.
+
+| Event                          | Explanation                                                                                 |
+| ------------------------------ | ------------------------------------------------------------------------------------------- |
+| PROVIDER_READY                 | The provider is ready to perform flag evaluations.                                          |
+| PROVIDER_ERROR                 | The provider signalled an error.                                                            |
```

Comment:
Signalling the provider is ready (PROVIDER_READY) with an event is something many vendors SDKs do: 

https://docs.launchdarkly.com/sdk/client-side/javascript#initializing-the-client
https://help.split.io/hc/en-us/articles/360020448791-JavaScript-SDK#basic-use

Especially in the browser, it's helpful to expose this information in an event because it gives application authors a means to know when they can start evaluating flags. Before this point, you might need to show loaders, spinners, etc.

For PROVIDER_CONFIGURATION_CHANGED, that's a way to _subscribe_ to configuration or flag values changes from the source of truth. This allows use to attach handlers which react to changes in flag values without the user performing any action (in the playground demo, the color of the app changes, for instance, without any refresh or polling). Many SDKs also support things like this:

https://docs.launchdarkly.com/sdk/features/flag-changes
https://docs.flagsmith.com/clients/javascript#api-reference (see `onChange` callback)

It's not really about what information these events provide over the evaluation details, as much as enabling different programming paradigms.

Reply:
> Ready and change make sense. I don't understand how someone would use error.

The use case I had in mind was particularly relevant for some of the SDKs/providers that maintain real-time connections. For example, flagd maintains a gRPC connection that could fail, and [Flagsmith has an "onError" hanlder](https://docs.flagsmith.com/clients/javascript#api-reference), and split has an event called `client.Event.SDK_READY_TIMED_OUT`. I was thinking it could be useful to have an error event which these sorts of thigns could be mapped to.